adagrams/game.py

```diff
@@ -1,13 +1,148 @@
 from random import randint
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
 
 def draw_letters():
-    pass
+    letter_bank = []
+    available_letters = build_letters_and_availability_list(LETTER_POOL)
+    
+    while len(letter_bank) < 10:
+        index_of_letter = get_index_of_letter()
 
+        if not available_letters[index_of_letter]['is_not_available']:
+            letter_bank.append(available_letters[index_of_letter]['letter'])
+            available_letters[index_of_letter]['is_not_available'] = True
+
+    reset_available_letters(available_letters)
+
+    return letter_bank
+
+def build_letters_and_availability_list(letter_pool):
+    available_letters = []
+    
+    for letter, letter_distribution in letter_pool.items():
+        for single_letter in range(letter_distribution):
+            available_letters.append({'letter':letter, 'is_not_available':False})
+    
+    return available_letters
+     
 def uses_available_letters(word, letter_bank):
-    pass
+    letters_in_word = list(word.upper())
+    is_valid = False
+
+    for letter in word:
+        same_letter_count = letters_in_word.count(letter) <= letter_bank.count(letter)
+        
+        if word.count(letter) > 0 and same_letter_count:
+            continue
+        else:
+            return is_valid
+    
+    is_valid = True
+
+    return is_valid
 
 def score_word(word):
-    pass
+    SCORE_CHART = {
+        1 : ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'],
+        2 : ['D', 'G'],
+        3 : ['B', 'C', 'M', 'P'],
+        4 : ['F', 'H', 'V', 'W', 'Y' ],
+        5 : ['K'],
+        8 : ['J', 'X' ],
+        10 : ['Q', 'Z']
+    }
+    BONUS_POINTS = 8
+    score = 0
+    word_case_insensitive = word.upper()
+
+    for letter in word_case_insensitive:
```

Comment:
We can directly call `upper` on the word we're looping over, like:
```suggestion
    for letter in word.upper():
```